<launch>
  <arg name="mdl_path" />
  <arg name="point_cloud_path" />
  <param name="robot_description" command="$(find xacro)/xacro $(find franka_description)/robots/panda/panda.urdf.xacro hand:=true" />
  <node pkg="mpinets_ros" type="planning_node.py" name="mpinets_planning_node" output="screen" clear_params="true">
    <param name="mdl_path" value="$(arg mdl_path)" />
    <param name="point_cloud_path" value="$(arg point_cloud_path)" />
  </node>
  <node name="static_link0_future_robot_states_link0_transform" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 panda_link0 planned_robot_states/panda_link0 100"  output="screen" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" args="joint_states:=/mpinets/joint_states" respawn="true" output="screen" />
  <node name="planned_robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" args="joint_states:=/mpinets/planned_joint_states" respawn="true" output="screen">
    <param name="/tf_prefix" value="/planned_robot_states" />
  </node>
  <node pkg="mpinets_ros" type="interaction_node.py" name="mpinets_interface" output="screen" clear_params="true" />
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find mpinets_ros)/rviz/config.rviz" />
</launch>
